### PR TITLE
Fix (ZEB-728) username validation

### PR DESCRIPTION
**Pull request by MitskevichVS**

### CODE REVIEW

File: client/app/_user/profile.html [hunk 1]
```diff
@@ -82,8 +83,14 @@
                 <div class="profile__form-row" layout="row">
                     <md-input-container flex class="_gray-bg">
                         <label>First Name</label>
-                        <input id="userFirstName" type="text" ng-model="$ctrl.user.firstName" name="firstName"
-                            ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="userFirstName"
+                            type="text"
+                            ng-model="$ctrl.user.firstName"
+                            name="firstName"
+                            ng-minlength="$ctrl.getValidationValue('name', 'minlength')"
```
Comment (danymill) [inline client/app/_user/profile.html]: Please, use one-time binding `::` (see details [here](https://docs.angularjs.org/guide/expression))

File: client/app/_user/profile.html [hunk 2]
```diff
@@ -82,8 +83,14 @@
...
+                            ng-maxlength="$ctrl.getValidationValue('name', 'maxlength')"
+                            ng-pattern="/^[a-zA-Z]+[a-zA-Z.-\d]+$/"
```
Comment (danymill) [inline client/app/_user/profile.html]: Move pattern into service as well

File: client/app/_user/profile.html [hunk 3]
```diff
@@ -96,8 +103,14 @@
...
                         <label>Last Name</label>
-                        <input id="userLastName" type="text" ng-model="$ctrl.user.lastName" name="lastName"
-                               ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="userLastName"
+                            type="text"
+                            ng-model="$ctrl.user.lastName"
+                            name="lastName"
+                            ng-minlength="$ctrl.getValidationValue('name', 'minlength')"
+                            ng-maxlength="$ctrl.getValidationValue('name', 'maxlength')"
+                            ng-pattern="/^[a-zA-Z]+[a-zA-Z.-\d]+$/"
```
Comment (danymill) [inline client/app/_user/profile.html]: See comments above

File: client/app/_users/modals/create_modal.html [hunk 4]
```diff
@@ -32,14 +32,19 @@ <h2 id="modalTitle">New user</h2>
                 <div layout layout-sm="column">
                     <md-input-container flex>
                         <label>First name</label>
-                        <input id="firstName" type="text" ng-model="user.firstName" name="firstName"
-                               ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="firstName"
+                            type="text"
+                            ng-model="user.firstName"
+                            name="firstName"
+                            ng-minlength="UtilService.getValidationValue('name', 'minlength')"
```
Comment (danymill) [inline client/app/_users/modals/create_modal.html]: See comments above

File: client/app/_users/modals/create_modal.html [hunk 5]
```diff
@@ -48,14 +53,20 @@ <h2 id="modalTitle">New user</h2>
                 <div layout layout-sm="column">
                     <md-input-container flex>
                         <label>Last Name</label>
-                        <input id="lastName" type="text" ng-model="user.lastName" name="lastName"
-                               ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="lastName"
+                            type="text"
+                            ng-model="user.lastName"
+                            name="lastName"
+                            ng-minlength="UtilService.getValidationValue('name', 'minlength')"
```
Comment (danymill) [inline client/app/_users/modals/create_modal.html]: See comments above

File: client/app/_users/modals/edit_modal.html [hunk 6]
```diff
@@ -21,8 +21,14 @@ <h2 id="modalTitle">Edit user</h2>
...
                         <label>First name</label>
-                        <input id="firstName" type="text" ng-model="user.firstName" name="firstName"
-                               ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="firstName"
+                            type="text"
+                            ng-model="user.firstName"
+                            name="firstName"
+                            ng-minlength="UtilService.getValidationValue('name', 'minlength')"
+                            ng-maxlength="UtilService.getValidationValue('name', 'maxlength')"
+                            ng-pattern="/^[a-zA-Z]+[a-zA-Z.-\d]+$/"
```
Comment (danymill) [inline client/app/_users/modals/edit_modal.html]: See comments above

File: client/app/_users/modals/edit_modal.html [hunk 7]
```diff
@@ -37,8 +43,14 @@ <h2 id="modalTitle">Edit user</h2>
                 <div layout layout-sm="column">
                     <md-input-container flex>
                         <label>Last Name</label>
-                        <input id="lastName" type="text" ng-model="user.lastName" name="lastName"
-                               ng-minlength="2" ng-maxlength="50" ng-pattern="/^[a-zA-Z]+$/"/>
+                        <input id="lastName"
+                            type="text"
+                            ng-model="user.lastName"
+                            name="lastName"
+                            ng-minlength="UtilService.getValidationValue('name', 'minlength')"
```
Comment (danymill) [inline client/app/_users/modals/edit_modal.html]: See comments above

**Review by danymill — CHANGES_REQUESTED** [review]

File: client/app/_user/profile.html (hunk 1 above)
Comment (MitskevichVS) [reply client/app/_user/profile.html]: fixed

File: client/app/_user/profile.html (hunk 2 above)
Comment (MitskevichVS) [reply client/app/_user/profile.html]: fixed

File: client/app/_user/profile.html (hunk 3 above)
Comment (MitskevichVS) [reply client/app/_user/profile.html]: fixed

File: client/app/_users/modals/create_modal.html (hunk 4 above)
Comment (MitskevichVS) [reply client/app/_users/modals/create_modal.html]: fixed

File: client/app/_users/modals/edit_modal.html (hunk 6 above)
Comment (MitskevichVS) [reply client/app/_users/modals/edit_modal.html]: fixed

File: client/app/_users/modals/create_modal.html (hunk 5 above)
Comment (MitskevichVS) [reply client/app/_users/modals/create_modal.html]: fixed

File: client/app/_users/modals/edit_modal.html (hunk 7 above)
Comment (MitskevichVS) [reply client/app/_users/modals/edit_modal.html]: fixed

File: client/app/_services/util.service.js [hunk 8]
```diff
@@ -93,21 +94,30 @@
...
                 }
             ]
         };
 
         return service;
 
+        function getValidationValue(objName, propName) {
+            const validationsArray = service.validations[objName];
+
+            return validationsArray.find(item => item.name === propName).value || null;
```
Comment (danymill) [inline client/app/_services/util.service.js]: if an item with `propName` is not found your code will throw an error

**Review by danymill — CHANGES_REQUESTED** [review]

**Review by danymill — APPROVED** [review]

**Pull request merged**